bigquery/test/tables.test.js

```diff
@@ -0,0 +1,213 @@
+        console.log.calledWith('PollExportJob: job status: %s', example.mocks.metadata.status.state),
+        'job status was reported'
+      );
+    });
+
+    it('should error if a job is not finished', function () {
+      var example = getSample();
+      var callback = sinon.stub();
+
+      var pendingState = { status: { state: 'PENDING' } };
+      example.mocks.job.getMetadata = sinon.stub().callsArgWith(0, null, pendingState);
+      example.program.pollExportJob(example.jobId, callback);
+
+      assert(example.mocks.bigquery.job.calledOnce, 'bigquery.job called once');
+      assert(example.mocks.job.getMetadata.calledOnce, 'getMetadata called once');
+      assert(callback.calledOnce, 'callback called once');
+      assert.equal(callback.firstCall.args.length, 1, 'callback received 1 argument');
+      assert.deepEqual(callback.firstCall.args[0], new Error('Job %s is not done'),
+        'callback received error'
+      );
+
+      assert(console.log.calledWith('PollExportJob: job status: %s', pendingState.status.state),
+        'job status was reported'
+      );
+
+      var doneState = { status: { state: 'DONE' } };
+      example.mocks.job.getMetadata = sinon.stub().callsArgWith(0, null, doneState);
+      example.program.pollExportJob(example.jobId, callback);
+
+      assert(example.mocks.bigquery.job.calledTwice, 'bigquery.job called a second time');
+      assert(example.mocks.job.getMetadata.calledOnce, 'new getMetadata called once');
+      assert(callback.calledTwice, 'callback called a second time');
+
+      assert.equal(callback.secondCall.args.length, 2, 'callback received 2 arguments');
+      assert.ifError(callback.secondCall.args[0], 'callback did not receive error');
+      assert.deepEqual(callback.secondCall.args[1], example.mocks.metadata,
+        'callback received metadata'
+      );
+
+      assert(console.log.calledWith('PollExportJob: job status: %s', doneState.status.state),
+        'job status was reported'
+      );
+    });
+
+    it('should handle error', function () {
+      var error = new Error('pollExportJobError');
+      var example = getSample();
+      var callback = sinon.stub();
+      example.mocks.job.getMetadata = sinon.stub().callsArgWith(0, error);
+      example.program.pollExportJob(example.jobId, callback);
+
+      assert(example.mocks.bigquery.job.calledOnce, 'bigquery.job called once');
+      assert(example.mocks.job.getMetadata.calledOnce, 'getMetadata called once');
+      assert(callback.calledOnce, 'callback called');
+      assert.equal(callback.firstCall.args.length, 1, 'callback received 1 argument');
+      assert.equal(callback.firstCall.args[0], error, 'callback received error');
+    });
+  });
+
+  describe('main', function () {
+    it('should call exportTableToGCS', function () {
+      var program = getSample().program;
+      program.exportTableToGCS = sinon.stub();
+
+      program.main(['export', 'bucket', 'file', 'dataset', 'table', 'JSON']);
```

Comment:
If any of these magic strings appear elsewhere in this file, create a single var at the top of the file that you can re-use throughout the file.